SpatialGDK/Source/SpatialGDKEditor/Private/SnapshotGenerator/SpatialGDKEditorSnapshotGenerator.cpp

```diff
@@ -142,23 +142,28 @@ bool CreateGlobalStateManager(Worker_SnapshotOutputStream* OutputStream)
 	return Worker_SnapshotOutputStream_WriteEntity(OutputStream, &GSM) != 0;
 }
 
-bool CreatePlaceholders(Worker_SnapshotOutputStream* OutputStream)
+bool CreatePlaceholders(Worker_SnapshotOutputStream* OutputStream, Worker_EntityId& NextAvailableEntityID)
 {
 	// Set up grid of "placeholder" entities to allow workers to be authoritative over _something_.
-	int PlaceholderCount = SpatialConstants::PLACEHOLDER_ENTITY_ID_LAST - SpatialConstants::PLACEHOLDER_ENTITY_ID_FIRST + 1;
-	int PlaceholderCountAxis = static_cast<int>(sqrt(PlaceholderCount));
-	checkf(PlaceholderCountAxis * PlaceholderCountAxis == PlaceholderCount, TEXT("The number of placeholders must be a square number."));
-	checkf(PlaceholderCountAxis % 2 == 0, TEXT("The number of placeholders on each axis must be even."));
 	const float CHUNK_SIZE = 5.0f; // in SpatialOS coordinates.
-	int PlaceholderEntityIdCounter = SpatialConstants::PLACEHOLDER_ENTITY_ID_FIRST;
-	for (int x = -PlaceholderCountAxis / 2; x < PlaceholderCountAxis / 2; x++)
+
+	const TArray<FName>& ServerWorkerTypeNames = GetDefault<USpatialGDKSettings>()->WorkerTypes.Array();
+	TArray<FString> ServerWorkerTypes;
+	for (auto& WorkerType : ServerWorkerTypeNames)
+	{
+		ServerWorkerTypes.Add(WorkerType.ToString());
+	}
```

Comment:
wait do we just take a reference to `GetDefault<USpatialGDKSettings>()->WorkerTypes.Array()`, then iterate through and add these to another array but of FString? Seems the only place we use is line below to setup the WorkerRequirementSet, current code seems like a bit of overkill?

Reply:
What should I do instead?